File: src/main/java/com/example/first/ClientMain.java
package com.example.first;

import java.io.IOException;
import java.net.Socket;
import java.util.Formatter;

public class ClientMain {

    public static void main(String[] args) throws IOException {
        final Socket socket = new Socket("127.0.0.1", 8080);

        final Formatter formatter = new Formatter(socket.getOutputStream());

        formatter.format("THIS IS SIMPLE TEXT");
        formatter.flush();

        formatter.close();

        socket.close();
    }
}

File: src/main/kotlin/com/example/client/ClientMain.kt
package com.example.client

import java.net.Socket
import java.util.Formatter
import java.util.Scanner
import kotlin.concurrent.thread

private const val PORT = 8080
private const val IP = "localhost"

fun main() {
    val socket = Socket(IP, PORT);
    println("Connected to server!")

//    val inputStream = socket.getInputStream()

    val inputStream = socket.getInputStream()
    thread {
        val scanner = Scanner(inputStream)
        while (true) {
            if (scanner.hasNextLine()) {
//                    System.out.println(scanner.nextLine())
                println(scanner.nextLine())
            }
        }
    }

    val outputStream = socket.getOutputStream()
    val terminalInput = Scanner(System.`in`)
    val writer = outputStream.bufferedWriter()

    while (true) {
        writer.write(terminalInput.nextLine())
        writer.newLine()
        writer.flush()
    }

}